### ISSUE TITLE
ModelPartIO prints wrong numbers in MPI

### ISSUE THREAD

**Issue by philbucher** [KratosMultiphysics/Kratos#5863]
After replacing the prints in the `ModelPartIO` from `std::cout` to the Logger, the prints are no longer correct, it only prints the numbers on one rank

I tried fixing but it seems to be not trivial

Hence making an issue, also as a reminder

FYI @KratosMultiphysics/technical-committee @jcotela

**Comment by jcotela** [comment]
@philbucher I do not think the numbers were correct before, it is just that you could work out the total by adding them... Essentially one should print the result of `DataCommunicator.SumAll(local_value)` or whatever reduction operation is correct in each case (max, min,...)

**Comment by philbucher** [comment]
I agree @jcotela, the problem is that the data-comm is not available everywhere

Also I am not sure what we want to print, imagine the partitioner uses the IO, would we want to print the local or the global values?

**Comment by adityaghantasala** [comment]
For me printing the local values on each rank makes more sense than printing nothing in other ranks (except on rank 0). This is because the data-comm is not available there.

**Comment by pooyan-dadvand** [comment]
Sounds `MPIModelPartIO` and meanwhile is an overkill to have it only for logging...

Meanwhile, The communicator in Modelpart don't have the corresponding communicator data?

**Comment by pablobecker** [comment]
On behalf of @KratosMultiphysics/implementation-committee 

We think for now we can go with using DefaultCommunicator (MPI_COMM_WORLD) , making rank 0 print the info of the complete domain (using sumall)

In the future, we can see how can we deal with split communicators when they are available. The current priority should be to fix the incorrect behavior reported in this issue.

**Comment by rubenzorrilla** [comment]
> On behalf of @KratosMultiphysics/implementation-committee
> 
> We think for now we can go with using DefaultCommunicator (MPI_COMM_WORLD) , making rank 0 print the info of the complete domain (using sumall)
> 
> In the future, we can see how can we deal with split communicators when they are available. The current priority should be to fix the incorrect behavior reported in this issue.

Just mentioning that #5489 is approved and ready to be merged. I don't know if this may affect the discussion at hand.

**Comment by adityaghantasala** [comment]
After #5489, I still think we can use GetDefaultComm function here.
